conda_smithy/templates/circle.yml.tmpl

```diff
@@ -9,6 +9,13 @@ general:
 {% endblock -%}
 {% block env_dependencies -%}
 {%- if matrix -%}
+checkout:
+  post:
+    # Checkout the merged PR for testing as CircleCI will just use the PR head otherwise.
+    - if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then git fetch origin +refs/pull/$CIRCLE_PR_NUMBER/head:pr/$CIRCLE_PR_NUMBER/head ; fi
+    - if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then git fetch origin +refs/pull/$CIRCLE_PR_NUMBER/merge:pr/$CIRCLE_PR_NUMBER/merge ; fi
+    - if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then git checkout -qf pr/$CIRCLE_PR_NUMBER/merge ; fi
```

Comment:
Ideally we would also check for merge conflicts and error out if present. However, we can't really do that without knowing what the base branch is as it may not be `master`. If we know that [information]( https://discuss.circleci.com/t/add-environment-varible-circle-pr-base-branch/6189 ), we could do something like [this]( https://github.com/conda-forge/staged-recipes/pull/1486/files#diff-29944324a3cbf9f4bd0162dfe3975d88R16 ).